src/org/sosy_lab/java_smt/solvers/princess/PrincessRationalFormulaManager.java

```diff
@@ -142,7 +135,7 @@ protected IExpression multiply(IExpression number1, IExpression number2) {
 
   @Override
   protected IExpression divide(IExpression number1, IExpression number2) {
-    return Rationals.div(toType(number1), toType(number2));
+    return Rationals.divWithSpecialZero(toType(number1), toType(number2));
```

Comment:
What is special about zero in this case? Please document as comment or provide a test that shows the behaviour, if the test does not yet exist.